include/irr/core/math/aabb2dSIMDIntBase.h

```diff
@@ -0,0 +1,259 @@
+#ifndef __IRR_AABB_SIMD_INT_BASE_H_INCLUDED__
+#define __IRR_AABB_SIMD_INT_BASE_H_INCLUDED__
+
+#include "irr/core/math/irrMath.h"
+#include "irr/core/memory/memory.h"
+#include "irr/core/alloc/AlignedBase.h"
+#include "../include/vectorSIMD.h"
+
+#include <iostream>
+
+#define DWN_CAST_THIS_PTR static_cast<const CRTP<VECTOR_TYPE>*>(this)
+#define SIGN_FLIP_MASK_XY	core::vector3du32_SIMD(0x80000000, 0x80000000, 0x00000000, 0x00000000)
+
+namespace irr { namespace core {
+
+
+template<template<typename> typename CRTP, typename VECTOR_TYPE>
+class aabbSIMDBase : public AlignedBase<_IRR_VECTOR_ALIGNMENT>
+{
+	static_assert(
+		std::is_same<VECTOR_TYPE, core::vector2di32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector2du32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector2df_SIMD>::value,
+		"assertion failed: cannot use this type");
+
+public:
+	inline void addPoint(const VECTOR_TYPE& other)
+	{
+		DWN_CAST_THIS_PTR->addBox();
+	}
+
+	inline VECTOR_TYPE getExtent() const
+	{
+		return DWN_CAST_THIS_PTR->getMaxEdge() - static_cast<const CRTP<VECTOR_TYPE>*>(this)->getMinEdge();
+	}
+
+	//need operator/ for core::vector32_i<uint32_t> and core::vector32_i<uint32_t>
+	/*inline VECTOR_TYPE getCenter() const
+	{
+		return static_cast<const CRTP<VECTOR_TYPE>*>(this)->getMaxEdge() - static_cast<const CRTP<VECTOR_TYPE>*>(this)->getMinEdge();
+	}*/
+
+	//! Resets the bounding box to a one-point box.
+	/** \param x X coord of the point.
+	\param y Y coord of the point. */
+	inline void reset(const VECTOR_TYPE& point)
+	{
+		CRTP<VECTOR_TYPE> pointBox(point.xyxy());
+		internalBoxRepresentation= pointBox;
+	}
+
+	//! Repairs the box.
+	/** Necessary if for example MinEdge and MaxEdge are swapped. */
+	inline CRTP<VECTOR_TYPE> repair()
+	{
+		//
+		//but will it work for aabbSIMDf for sure?
```

Comment:
sure because `get*Edge` returns a normal natural 2d point and `reset` + `addPoint` should work on it just fine

Reply:
description of this function says: 
/** Necessary if for example MinEdge and MaxEdge are swapped. */

so in this situation if minEdge and maxEdge are switched, then getMinEdge will return maxEdge with fliped sign

Reply:
yes, but you need to anticipate `repair` being called on a totally valid aabb (that does not need switching)